src/impl/conv_im2col.jl

```diff
@@ -162,7 +162,7 @@ function ∇conv_data_im2col!(
                     col_ptr = pointer(col_slice)
                     gemm!(Val(false), Val(true), M, N, K, alpha, dy_ptr, w_ptr, T(0), col_ptr)
```

Comment:
While trying to determine how to simplify https://github.com/FluxML/NNlib.jl/pull/539/files#diff-92d81e5ae578721d9c48a17e46ad5e36ca65e58cd5f786e5789820f16d9bc9e7R306-R312, it occurred to me that we probably shouldn't be handling this in `col2im!`. `im2col!` doesn't handle `beta`, and `∇conv_filter_im2col!` does via https://github.com/FluxML/NNlib.jl/blob/v0.9.6/src/impl/conv_im2col.jl#L110. Hence, perhaps this would work instead:
```suggestion
                    gemm!(Val(false), Val(true), M, N, K, alpha, dy_ptr, w_ptr, beta, col_ptr)
```

Reply:
Unfortunately not, because the col2im sets dx to 0. Also the gemm! here is not the correct matrix to apply beta to.

Reply:
That makes sense. But then the way it's done now also doesn't work, because there's no `beta * grad` as mentioned in https://github.com/FluxML/NNlib.jl/issues/538. Of the 3 branches in https://github.com/FluxML/NNlib.jl/pull/539/files#diff-92d81e5ae578721d9c48a17e46ad5e36ca65e58cd5f786e5789820f16d9bc9e7R306-R312, only the first `fill!(...)` does anything?

Reply:
So the beta == 0 case, is optimized to fill since thats saying x = alpha * newvals. the beta = 1 case, can just be optimized to x += alpha * newvals, so we don't need any additional work since x already contains x * old(x). The alternate case just requires x *= beta, since then the final x will be x = beta * old(x) + newvals.

Reply:
So the alternate case should be `x .*= beta` then?

Reply:
Yes, looks like I missed that =

Reply:
Fixed (and added test) for that third case

Reply:
And with that fix it actually does look like a bunch of previously failing tests now pass.